accounts-db/src/account_locks.rs

```diff
@@ -0,0 +1,162 @@
+#[cfg(feature = "dev-context-only-utils")]
+use qualifier_attr::qualifiers;
+use {
+    ahash::{AHashMap, AHashSet},
+    log::debug,
+    solana_sdk::{pubkey::Pubkey, transaction::TransactionError},
+    std::collections::hash_map,
+};
+
+#[derive(Debug, Default)]
+pub struct AccountLocks {
+    write_locks: AHashSet<Pubkey>,
+    readonly_locks: AHashMap<Pubkey, u64>,
+}
+
+impl AccountLocks {
+    /// Lock the account keys in `keys` for a transaction.
+    /// The bool in the tuple indicates if the account is writable.
+    /// Returns an error if any of the accounts are already locked in a way
+    /// that conflicts with the requested lock.
+    pub fn try_lock_accounts_for_transaction<'a>(
```

Comment:
nitpick on naming: the function say do something "_for_transaction", but there is no transaction in parameter list, only account keys. So maybe just "try_lock_accounts()"?

Reply:
a8182f81db8d84f7e10ea017650d36d0c733e227